.github/workflows/hermetic_library_generation.yaml

```diff
@@ -24,24 +24,48 @@ jobs:
   library_generation:
     runs-on: ubuntu-latest
     steps:
+    - name: Prerequisite
+      shell: bash
+      run: |
+        if [[ "${GITHUB_REPOSITORY}" != "${REPO_FULL_NAME}" ]]; then
+          echo "This PR comes from a fork. Skip library generation."
+          exit 0
+        fi
     - uses: actions/checkout@v4
       with:
         fetch-depth: 0
         token: ${{ secrets.CLOUD_JAVA_BOT_TOKEN }}
+    - name: Install Maven modules
+      shell: bash
+      run: |
+        git checkout "${HEAD_REF}"
+        mvn install -B -ntp -DskipTests -Dclirr.skip -Dcheckstyle.skip
+      env:
+        HEAD_REF: ${{ github.head_ref }}
+    - name: Get local generator version
+      shell: bash
+      run: |
+        GENERATOR_VERSION=$(mvn help:evaluate -Dexpression=project.version -q -DforceStdout -pl gapic-generator-java)
```

Comment:
Can this step be combined with the step below?

Reply:
Done.